### ISSUE TITLE
Use DASK to support caching and parallelism

### ISSUE THREAD

**Issue by PGijsbers** [openml-labs/gama#3]
Perhaps gama could switch from evaluating multiple pipelines single-core in parallel to evaluating one pipeline multi-core sequentially. This could lead to improved performance due to e.g. less memory use.

**Comment by PGijsbers** [comment]
Besides the general pipelines, we currently train encoding/imputation models for every pipeline. While these steps are very quick, it is still wasteful.

**Comment by PGijsbers** [comment]
Closing this because I don't think DASK is the right tool, and the issue is a bit too broad. Will open a new issue to look into pipeline caching.
